homeassistant/components/kiosker/coordinator.py

```
@@ -0,0 +1,80 @@
+
+from .const import DOMAIN, POLL_INTERVAL
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class KioskerDataUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching data from the Kiosker API."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        api: KioskerAPI,
+        config_entry: ConfigEntry,
+    ) -> None:
+        """Initialize."""
+        self.api = api
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=timedelta(seconds=POLL_INTERVAL),
+            config_entry=config_entry,
+        )
+
+    async def _async_update_data(self) -> dict[str, Any]:
+        """Update data via library."""
+        try:
+            status = await self.hass.async_add_executor_job(self.api.status)
+            blackout = await self.hass.async_add_executor_job(self.api.blackout_get)
+            screensaver = await self.hass.async_add_executor_job(
+                self.api.screensaver_get_state
+            )
+        except (OSError, TimeoutError) as exception:
+            _LOGGER.warning(
+                "Connection failed for Kiosker: %s", exception, exc_info=True
+            )
+            raise UpdateFailed(exception) from exception
+        except Exception as exception:
+            # Check if this is an authentication error (401)
+            if self._is_auth_error(exception):
+                _LOGGER.warning("Authentication failed for Kiosker: %s", exception)
+                raise ConfigEntryAuthFailed("Authentication failed") from exception
+
+            _LOGGER.warning(
+                "Failed to update Kiosker data: %s", exception, exc_info=True
+            )
+            raise UpdateFailed(exception) from exception
+        else:
+            return {
+                "status": status,
+                "blackout": blackout,
+                "screensaver": screensaver,
+            }
+
+    def _is_auth_error(self, exception: Exception) -> bool:
```

can you raise a specific exception from the library. This is code that HA shouldn't be concerned about